src/sync/impl/apple/network_reachability_observer.cpp

```diff
@@ -0,0 +1,131 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2016 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#include "sync/impl/apple/network_reachability_observer.hpp"
+
+#if !TARGET_OS_WATCH
+
+using namespace realm;
+
+namespace {
+
+static NetworkReachabilityStatus reachability_status_for_flags(SCNetworkReachabilityFlags flags)
+{
+    if (!(flags & kSCNetworkReachabilityFlagsReachable)) {
+        return NotReachable;
+    }
+
+    NetworkReachabilityStatus status = NotReachable;
+
+    if (!(flags & kSCNetworkReachabilityFlagsConnectionRequired)) {
+        status = ReachableViaWiFi;
+    }
+
+    if ((flags & kSCNetworkReachabilityFlagsConnectionOnTraffic) &&
+        !(flags & kSCNetworkReachabilityFlagsInterventionRequired)) {
+        status = ReachableViaWiFi;
+    }
+
+#if TARGET_OS_IPHONE
+    if (flags & kSCNetworkReachabilityFlagsIsWWAN) {
+        status = ReachableViaWWAN;
```

Comment:
I'm not sure this logic is right. Above it looks like we're going out of our way to avoid treating `kSCNetworkReachabilityFlagsInterventionRequired` as indicating the connection is available, but we're not doing that here.

It seems like more appropriate logic would be to check whether a connection is required, and if it requires intervention (or won't be triggered due to our use of low-level socket APIs) then just return "not reachable". Then once we've established that a connection isn't required attempt to classify the connection into wifi vs cellular.